src/librustc/hir/lowering.rs

```diff
@@ -2484,37 +2456,35 @@ impl<'a> LoweringContext<'a> {
             };
 
             trace!("exist ty from async fn def index: {:#?}", exist_ty_def_index);
-            let exist_ty_id = this.generate_existential_type(
+            let _exist_ty_id = this.generate_existential_type(
                 exist_ty_node_id,
                 exist_ty_item,
                 span,
                 exist_ty_span,
             );
 
-            (exist_ty_id.node_id, lifetime_params)
+            (exist_ty_node_id, lifetime_params)
```

Comment:
Return `_exist_ty_id` from this block here instead of `exist_ty_node_id`. Then remove the `let exist_ty_hir_id` binding below, since that will be the same as the `HirId` returned here.